src/fclone.js

```diff
@@ -1,5 +1,13 @@
 'use strict';
 
+// see if it looks and smells like an iterable object, and do accept length === 0
+function isArrayLike(item) {
+  if (Array.isArray(item)) return true;
+
+  var len = item && item.length;
+  return typeof len === 'number' && (len === 0 || (len - 1) in item);
```

Comment:
I don't get the `(len -1) in item`. If Array length is < 0 it's not iterable, isn't it? IMO `return typeof len === 'number' && len > 0;` would suffice.